web_src/js/features/repo-editor.js

```diff
@@ -106,7 +106,7 @@ export function initRepoEditor() {
     if (e.keyCode === 191) {
       parts = $(this).val().split('/');
       for (let i = 0; i < parts.length; ++i) {
-        value = parts[i];
+        value = encodeURIComponent(parts[i]);
```

Comment:
I think we shouldn't do encoding here. 

We should do something like `$(...).html(value)`, instead of the $(`<span>${value}</span>`) in code.

Reply:
`$(...).html(value)` this would still allow to execute arbitrary things like `<iframe onload=alert(1)></iframe>`.

Reply:
Oh sorry, my typo. I mean `$(...).html(text)` 😂

Reply:
Oh my keyboard .... I mean `$(...).text(value)` 😭

Reply:
Well, `$(`<span class="section"><a href="#"></a></span>`).text(value)` will replace the content of outer `<span>`.

I will try to edit it online. Maybe we can use `$(`<span class="section"><a href="#">${encode(value)}</a></span>`)`. Just a minute.

Reply:
Well, we can not do `encodeURIComponent` for HTML text.

And since we do not import a HTML escaper function, so we can only to `$span.find('a').text(value)` ...

Sorry for so many comments.

Reply:
> we do not import a HTML escaper function

We have the escape-goat module for that, `import {htmlEscape} from 'escape-goat';` is used in other files. But using `.textContent` or jquery `.text` seems fine too.

Reply:
As long as it doesn't posses the ability to see the value as raw DOM(such that `<b>Hi</b>` will be literally inserted and seen as a bold text) it will be fine to fix the "bug".

Reply:
`htmlEscape` will be faster, so I have a slight preference for it. It can even be used as a tagged template. Some examples from node repl:

```js
> `<span class="section"><a href="#">${htmlEscape('<b>Hi</b>')}</a></span>`
'<span class="section"><a href="#">&lt;b&gt;Hi&lt;/b&gt;</a></span>'
> htmlEscape`<span class="section"><a href="#">${'<b>Hi</b>'}</a></span>`
'<span class="section"><a href="#">&lt;b&gt;Hi&lt;/b&gt;</a></span>'
```